src/main/java/org/springframework/data/elasticsearch/core/IndexCoordinates.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.core;
+
+import org.springframework.util.Assert;
+
+/**
+ * Value object encapsulating index name(s) and index type(s).
+ *
+ * @author Mark Paluch
+ * @author Christoph Strobl  @author Peter-Josef Meisch
+ */
+public class IndexCoordinates {
+
+	private final String[] indexNames;
+	private final String[] typeNames;
+
+	public IndexCoordinates(String[] indexNames) {
```

Comment:
Let's make constructors `private` so creation works only using static factory methods and mutators (`of(…)`, `withIndex(…)`, `withType(…)`).